proto/viam/app/data/v1/data.proto

```diff
@@ -234,6 +234,7 @@ message GetLatestTabularDataRequest {
   string part_id = 1;
   string resource_name = 2;
   string method_name = 3;
+  string resource_type = 4;
```

Comment:
would it make more sense to call this resource_subtype? according to the viam [glossary](https://docs.viam.com/appendix/glossary/#:~:text=type%20is%20component%20or%20service.%20subtype%20refers%20to%20a%20specific%20component%20or%20service%2C%20like%20a%20camera%20or%20vision.) resource type is either component or service and subtype is the actual component/service type like sensor or arm etc.

Reply:
Good callout! Changing this to `resource_subtype` instead